erp/views.py

```diff
@@ -864,27 +865,33 @@ def contrib_admin_infos(request):
 @create_revision(request_creates_revision=lambda x: True)
 def contrib_edit_infos(request, erp_slug):
     erp = get_object_or_404(Erp, slug=erp_slug)
-    initial = {"lat": float(erp.geom.y), "lon": float(erp.geom.x)}
+    initial = {"lat": Decimal(erp.geom.y), "lon": Decimal(erp.geom.x)}
     if request.user.is_authenticated and erp.user is request.user:
         libelle_next = schema.SECTION_A_PROPOS
         next_route = schema.SECTIONS[schema.SECTION_A_PROPOS]["edit_route"]
     else:
         libelle_next = schema.SECTION_TRANSPORT
         next_route = schema.SECTIONS[schema.SECTION_TRANSPORT]["edit_route"]
+
     if request.method == "POST":
-        form = forms.PublicErpEditInfosForm(request.POST, instance=erp)
+        form = forms.PublicErpEditInfosForm(request.POST, instance=erp, initial=initial)
         if form.is_valid():
-            erp = form.save()
-            if request.user.is_authenticated and erp.user is None:
-                erp.user = request.user
-                erp.save()
+            if form.has_changed():
+                erp = form.save()
+                if request.user.is_authenticated and erp.user is None:
+                    erp.user = request.user
+                    erp.save()
             if erp.activite.slug == "autre":
```

Comment:
this can be under the `form.has_changed` block I guess, and so, we could have `if form.is_valid() and form.has_changed()`

Reply:
I did what I could to take this into account, but we also need to redirect the user on the next step